src/chocolatey.resources/helpers/functions/Update-SessionEnvironment.ps1

```diff
@@ -56,31 +56,34 @@ None
   }
 
   if ($refreshEnv) {
-    Write-Output "Refreshing environment variables from the registry for powershell.exe. Please wait..."
+    Write-Output 'Refreshing environment variables from the registry for powershell.exe. Please wait...'
   } else {
-    Write-Verbose "Refreshing environment variables from the registry."
+    Write-Verbose 'Refreshing environment variables from the registry.'
   }
 
   $userName = $env:USERNAME
   $architecture = $env:PROCESSOR_ARCHITECTURE
   $psModulePath = $env:PSModulePath
 
-  #ordering is important here, $user comes after so we can override $machine
-  'Process', 'Machine', 'User' |
-    % {
-      $scope = $_
-      Get-EnvironmentVariableNames -Scope $scope |
-        % {
-          Set-Item "Env:$($_)" -Value (Get-EnvironmentVariable -Scope $scope -Name $_)
+  #ordering is important here, $user should override $machine...
+  $Scopes = 'Process', 'Machine'
+  if (($userName -ne 'SYSTEM') -and ($userName -ne "$($env:COMPUTERNAME)`$")) {
+    # but only if not running as the SYSTEM/machine in which case user can be ignored.
+    $Scopes += 'User'
+  }
+  foreach ($Scope in $Scopes) {

```

Comment:
I generally recommend avoiding plural variables be used alongside singular counterparts; it's very easy to mess things up with a single-character typo and not notice that the logic is totally broken until later. Maybe use `$ScopeList` instead for the latter variable?